### PR TITLE
Develop fix manifest generator

### PR DESCRIPTION
**Pull request by linglp**
I updated the following: 
* For "Check List", changed it from "list" to "list strict" 
* "check list" now takes the following valid values: `ab, cd, ef, gh`
* For attribute "Check Regex List", change "list::regex match [a-f]" to "list strict::regex match [a-f]" 
* Add an additional attribute in data model: Check Regex Format
* Update invalid_test_manifest.csv and valid_test_manifest.csv
* Update example.model.jsonld
 
In addition, I also updated the ManifestGenerator class so that something like `elif "list" in validation_rules and values` gets changed to `elif ("list strict" in validation_rules) or ("list like" in validation_rules) and valid_values`

When I created the manifest based on the example data model and opened it in Firefox (see an example here: https://docs.google.com/spreadsheets/d/1Ax_fb-TdQK67i9B60rncjAffVYJ8_hNKpguXBuIEH3k/edit#gid=0), I could now see a list of valid values for "Check List" in sheet 2. 

> But when I tried to enter a comma separated list under column "Check List" in sheet 1, I still got an error message related to the data validation rules. Should we update the "Data validation" rule as "On invalid data: show warning" instead of "Reject input" so that a comma separate list could show up?

^^ Note: In my code, for list in validation rule that allows multi-selection, I set "strict" to "False" so that users could enter multiple values. See here: https://github.com/Sage-Bionetworks/schematic/pull/847/files#:~:text=%3DNone%2C-,strict%3DFalse%2C,-custom_ui%3DFalse

> If we can't change all the "list" to "list like"/"list strict" in all existing data models, should we still maintain the logic related to "list"?

^^ Note: all the old logic related to "list" in schemaGenerator was handled by branch `develop-listlike-behavior`. After merging that branch with this branch here, there seems to be no more test errors.

### CODE REVIEW

**Comment by linglp** [comment]
For testing, here's my test manifest: https://docs.google.com/spreadsheets/d/1DwWA_f6kYwPkLlM0jpHuTAyltk8CuWS9nyxrGBcq8oY/edit#gid=0

*Note*: 
* "Check List" could now accept a list of values (separate by commas)
* If the value that you enter in "Check Regex Format" does not match [a-f], it will reject your entry. Otherwise, you would see the text turns red and then turn black (which is expected based on the formatting rules set in manifest generator)

**Comment by linglp** [comment]
This PR is related to #846

**Comment by GiaJordan** [comment]
@GiaJordan I'll update the documentation for google sheets limitations with regex matching in combo with list rule

File: tests/test_validation.py [hunk 1]
```diff
@@ -100,6 +100,15 @@ def test_invalid_manifest(self,helpers,sg,metadataModel):
             invalid_entry = 'ab cd ef'
             ) in errors
 
+        assert GenerateError.generate_regex_error(
```
Comment (GiaJordan) [inline tests/test_validation.py:103]: Can you add this assertion to `test_in_house_validation` as well?

**Review by GiaJordan — CHANGES_REQUESTED** [review]

**Review by GiaJordan — APPROVED** [review]
🚀

**Pull request merged**